pkg/gentrait/securityevent/model.go

```diff
@@ -0,0 +1,87 @@
+package securityevent
+
+import (
+	"context"
+	"strconv"
+	"sync"
+	"time"
+
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/smart-core-os/sc-api/go/types"
+	"github.com/smart-core-os/sc-golang/pkg/resource"
+	"github.com/vanti-dev/sc-bos/pkg/gen"
+)
+
+var subSystems = [2]string{"access control", "cctv"}
+
+type Model struct {
+	lastSecurityEvent *resource.Value // of *gen.SecurityEvent
+	allSecurityEvents []*gen.SecurityEvent
+	genId             int
+	mu                sync.Mutex
```

Comment:
Generally with Mutexes (and other sync types) we put it before the things that it guards and comment what it guards:
```go
type Foo struct {
  other Prop

  mu sync.Mutex // guards the props below, or guards a and b, or guards non-exported fields, or something like that
  a string
  b int

  other Prop
}
```